src/components/molecules/NavSearchBar/NavSearchBar.tsx

```diff
@@ -170,6 +173,7 @@ const NavSearchBar = () => {
       </div>
 
       <InputField
+        value={searchInputValue}
         inputClassName="NavSearchBar__search-input"
         onChange={onSearchInputChange}
         placeholder="Search for people, rooms, events..."
```

Comment:
I don't like the fact that we have 2 states(UI/functional) of the same thing. It doesn't seem right to me. searchQuery should derive & be dependent from searchInputValue. We have react-hook-form to handle the basic input state management. It has nice API, which includes `reset ` and `watch`. Then we can move setting search query into useEffect, dependent on that watch value `src/components/molecules/Chatbox/Chatbox.tsx`

Reply:
I gues you don't need handelSubmit, maybe there is a useful `mode` that suits us that can be found in https://react-hook-form.com/

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1348/commits/270c5e913961b67432b9e5cc8d2096c756b6203c